### ISSUE TITLE
[jvm] awkward main class naming with module-level main

### ISSUE THREAD

**Issue by Gama11** [HaxeFoundation/haxe#9609]
If you want to use `java -cp`, the name class name must be given explicitly, and that's quite awkward to use with a module-level `main()` field because of the naming:

```
java -cp <stuff> _Main.Main_Fields_
```

Could we do something about that, e.g. have a special case for entry points and simply generate it as `Main`?

(for `-jar` the main name from the manifest is implied, but you can't use `-jar` and `-cp` at the same time)

**Comment by Simn** [comment]
I don't think we should introduce special cases here... if you want nicer class names, you can always just use a class.

**Comment by Gama11** [comment]
Hm, well maybe there is a solution without special cases for `main()`. Why generate this as `_Fields_` at all if there are no other types in the `Main` module? It could just end up in `Main.class` to begin with.
